tests/test_measurements.py

```diff
@@ -60,6 +60,25 @@ def test_ObservableReturnTypes(return_type, value):
     assert isinstance(return_type, qml.measurements.ObservableReturnTypes)
     assert repr(return_type) == value
 
+def test_expval_non_commuting_observables():
+    """Test expval with multiple non-commuting operators """
+    dev = qml.device("default.qubit", wires=5)
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.Hadamard(1)
+        qml.Hadamard(0)
+        qml.PauliZ(0)
+        qml.Hadamard(3)
+        return [qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
+        qml.expval(qml.PauliX(0)),
+        qml.expval(qml.PauliZ(1)),
+        qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
+        qml.expval(qml.PauliX(3))]
+
+    assert all(np.isclose(circuit(),np.array([0., -1., 0., 0., 1.])))
```

Comment:
@Qottmann I recommend doing as you suggest, and moving this test to  is there any benefit to adding more tests to a file `tests/transform/test_split_non_commuting.py`.

I would also recommend adding more tests, yes. In particular:

- It would be good to have some unit tests. That is, tests that ensure that the transform works correctly at the _unit_ level. So testing that various input tapes produce the expected output tapes:

  - Those with non-commuting observables are not split
  - Those with commuting observables are split into the right number of groups
  - Tapes with variances/samples work

- In addition, it would be good to add some integration tests, to ensure the `device` changes are working nicely (e.g., like the test you have above!). Some good tests to add include:

  - Testing that the pipeline works when using TF/Torch/JAX
  - Testing that the pipeline works when _differentiating_ using TF/Torch/JAX

  As you have probably noticed, the processing function can be hard to get working in a differentiable manner!